include/wex/Engine.hpp

```diff
@@ -30,22 +32,22 @@ class Game {
 
 	/// \brief The draw function is resposible for rendering everything that is visible
 	/// on the screen.
-	/// \param graphics A Graphics Controller that belongs to the engine. the controller contains
-	///	       member functions that can help draw shapes, lines and images to the screen.
-	virtual void draw(GraphicsController& graphics) = 0;
+	virtual void draw() = 0;
 
 	virtual ~Game() = default;
 
 	inline void setGraphicsController(GraphicsController& controller) noexcept {
-		this->g = std::unique_ptr<GraphicsController>(&controller);
+		this->g = &controller;
 	}
 
-	inline const util::NotNullPtr<GraphicsController> graphicsController() const noexcept {
-		return util::NotNullPtr(this->g.get());
+	[[nodiscard]] inline NotNullPtr<GraphicsController> const graphicsController() const noexcept {
```

Comment:
This can just be a reference IMO, `not_null` is useful for receiving params (notifies caller that the address is stored etc and used later) and as members.